### PR TITLE
fix(core): set totp qrcode service and user name

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Improve the QRCode of TOTP: add service name tag and user name tag.

This also fix the scan failure in some devices.

## Testing

Local tested.

image

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/interaction/utils/user-display-name.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { type User } from '@logto/schemas';
+
+/**
+ * Get user display name from multiple fields
+ */
+export const getUserDisplayName = ({
```
Comment (charIeszhao) [inline packages/core/src/routes/interaction/utils/user-display-name.ts:6]: nit: maybe we should put it in `@logto/shared` package

File: packages/core/src/routes/interaction/additional.ts [hunk 2]
```diff
@@ -128,10 +139,35 @@ export default function additionalRoutes<T extends IRouterParamContext>(
...
-      ctx.body = {
-        secret,
-        secretQrCode: await qrcode.toDataURL(`otpauth://totp/?secret=${secret}`),
-      };
+      if (isRegisterInteractionResult(profileVerifiedInteraction)) {
+        const {
+          username = null,
+          primaryEmail = null,
+          primaryPhone = null,
+          name = null,
```
Comment (simeng-li) [inline packages/core/src/routes/interaction/additional.ts:147]: Looks like these values could remain undefined.

Comment (wangsijie) [reply packages/core/src/routes/interaction/additional.ts:147]: It won't.

image

File: packages/core/src/routes/interaction/utils/user-display-name.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/utils/user-display-name.ts:6]: updated

**Review by charIeszhao — APPROVED** [review]

File: packages/shared/src/utils/user-display-name.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Get user display name from multiple fields
+ */
+export const getUserDisplayName = ({
```
Comment (charIeszhao) [inline packages/shared/src/utils/user-display-name.ts:4]: Can you also update the usage in `console` to make sure both console and core have the same reference?

**Review by simeng-li — APPROVED** [review]

**Pull request merged**